src/utils/crvoting-utils.js

```diff
@@ -42,28 +44,40 @@ const outcomeStringMapping = {
   [OUTCOMES.InFavor]: 'Voted in favor',
 }
 
-const appealRulingStringMapping = {
-  [OUTCOMES.Leaked]: 'Invalid ruling',
-  [OUTCOMES.Refused]: 'Refused',
-  [OUTCOMES.Against]: 'Ruled against',
-  [OUTCOMES.InFavor]: 'Ruled in favor',
-}
-
 export function juryOutcomeToString(outcome) {
   if (!outcome) {
     return outcomeStringMapping[OUTCOMES.Refused]
   }
   return outcomeStringMapping[outcome]
 }
 
+const appealRulingStringMapping = {
+  [OUTCOMES.Leaked]: 'Invalid ruling',
+  [OUTCOMES.Refused]: 'Refused',
+  [OUTCOMES.Against]: 'Ruled against',
+  [OUTCOMES.InFavor]: 'Ruled in favor',
+}
+
 export function appealRulingToString(outcome, confirm) {
   if (!outcome) {
     return confirm ? NOBODY_CONFIRMED : NOBODY_APPEALED
   }
   return appealRulingStringMapping[outcome]
 }
 
-const VALID_OUTCOMES = [OUTCOMES.Refused, OUTCOMES.Against, OUTCOMES.InFavor]
+const finalRulingStringMapping = {
+  [OUTCOMES.Leaked]: 'Invalid ruling',
+  [OUTCOMES.Refused]: 'Refused',
+  [OUTCOMES.Against]: 'Ruling against',
+  [OUTCOMES.InFavor]: 'Ruling in favor',
+}
+
+export function finalRulingToString(outcome) {
+  if (!outcome) {
+    return finalRulingStringMapping[OUTCOMES.refused]
```

Comment:
Interesting; can a final ruling be refused?

Reply:
Yes, in the case that the majority refuses to vote, or nobody actually votes.